### PR TITLE
Run tests locally

### LINKED ISSUE #34: There should be an easy way to run the build checks locally
When checking a pull request, Travis runs `bundle exec htmlproof --href-swap ".*pyconuk.org/:/" _site/`.

It should be possible to run this locally, easily, both if you're running inside docker or if you're not.

Assigning to @ghickman cos he seems to enjoy this kind of thing, but don't feel you've got to take it on!

### PR DESCRIPTION
**Pull request by ghickman**
#### What's this PR do?

Introduce the `make tests` command for running the tests locally. Both tests (name linting and html proofing) now live in the `tests` directory.

I've also fixed a bug in the name linting test to do with single brackets conditional and added the phony stuff to the makefile to avoid issues with the tests directory and tests target.
#### How should this be manually tested?

Checkout this branch and run `make tests`.
#### What are the relevant tickets?

Fixes #34

### CODE REVIEW

**Comment by ghickman** [comment]
Could someone check this locally please? I'm not sure if the name, `evil_hoover`, is a locally generated name for my docker container or not.

**Comment by d0ugal** [comment]
```
/tmp/pyconuk.github.io run-tests-locally  
❯ make tests
Checking the conference name
Proofing the HTML
Post http:///var/run/docker.sock/v1.17/containers/evil_hoover/start: dial unix /var/run/docker.sock: no such file or directory. Are you trying to connect to a TLS-enabled daemon without TLS?
FATA[0000] Error: failed to start one or more containers 
make: *** [tests] Error 1
```

**Comment by d0ugal** [comment]
... That error may be related to my setup actually, I can't run the website at all. I've not used docker for a while on this machine.

File: Makefile [hunk 1]
```diff
@@ -12,3 +13,9 @@ build:
 
 run:
 	docker run -p 80:4000 -v $(pwd):/opt/pyconuk/site:rw -ti pyconuk.org
+
+tests:
+	@echo "Checking the conference name" && bash tests/name-lint.sh
+	@echo "Proofing the HTML" && docker start evil_hoover && docker exec evil_hoover tests/htmlproof.sh
```
Comment (inglesp) [inline Makefile:19]: What's `evil_hoover` here?

**Comment by inglesp** [comment]
Thanks for working on this @ghickman.

This doesn't actually work for me either:

```
$ make tests
Checking the conference name
Proofing the HTML
Error response from daemon: No such container: evil_hoover
FATA[0000] Error: failed to start one or more containers 
make: *** [tests] Error 1
```

I think that the problem is that you're trying to start a container based on a randomly-assigned name, "evil_hoover".  I don't know enough about docker to know what the right thing to do here is.

Also, `name-lint.sh` isn't executable at the moment, which means that you can't run it from the command line without `chmod`ding it.

**Comment by inglesp** [comment]
Closing, since this is now working at https://github.com/PyconUK/pyconuk.org.

**Pull request closed without merging**